services/tracing/tracing/application.go

```diff
@@ -28,14 +31,16 @@ type Service struct {
 
 func (s *Service) HandlePipelineMessage(pipelineMessage pipeline.Message, time time.Time) error {
 	if len(pipelineMessage.PipelineSteps)-(int(pipelineMessage.StepIndex+1)) < 0 {
-		return fmt.Errorf("steps remaining cannot be smaller than 0 (pipelinesteps len: %d, stepindex: %d)",
+		return fmt.Errorf("%w: steps remaining cannot be smaller than 0 (pipelinesteps len: %d, stepindex: %d) for id %s",
+			ErrInvalidStepsRemaining,
```

Comment:
what is the purpose of this exported error? Don't see it used anywhere

Reply:
I thought I changed it in the tests. The test failed because the string of the error changed, so I created an error variable, wrapped it and used the `errors.Is` to validate the test.